modules/core/src/ocl.cpp

```diff
@@ -113,6 +113,14 @@
 
 #include "opencv2/core/opencl/runtime/opencl_core.hpp"
 
+#ifdef HAVE_DIRECTX
+namespace cv { namespace directx {
+struct OpenCLDirectXImpl;
+OpenCLDirectXImpl* createDirectXImpl();
+void deleteDirectXImpl(OpenCLDirectXImpl**);
```

Comment:
Please move this into `modules/core/src/directx.cpp` and use it instead (here and in directx.hpp).

Consider using this namespace:
- `cv::directx::internal`
- or `cv::directx::detail`

Reply:
OK, move this into `modules/core/src/directx.cpp` and using `cv::directx::internal` namespace.